development_extensions/oni-dev-extension/extension.js

```diff
@@ -156,6 +160,27 @@ function activate(context) {
     };
     let disposable = vscode.workspace.registerTextDocumentContentProvider('foo', textContentProvider)
     disposable.dispose();
+
+    // Configuration
+
+    const rlsLocation = vscode.workspace.getConfiguration().get("reason_language_server.location");
+    console.error("Configured RLS location: ", rlsLocation);
+
+    const editorFontFamily = vscode.workspace.getConfiguration().get("editor.fontFamily");
+    console.error("Editor Font Family: ", editorFontFamily);
+
+    const dontCrash = vscode.workspace.getConfiguration().get("developer.oni.dont-crash");
+    console.error("Don't crash: ", dontCrash);
+
+    vscode.workspace.onDidChangeConfiguration(event => {
+      if (event.affectsConfiguration("developer.oni.dont-crash")) {
+        const dontCrash = vscode.workspace.getConfiguration().get("developer.oni.dont-crash");
+        console.error("Setting changed, Don't crash: ", dontCrash);
+      } else {
+        const dontCrash = vscode.workspace.getConfiguration().get("developer.oni.dont-crash");
+        console.error("Setting did not change, Don't crash: ", dontCrash);
+      }
+    });
```

Comment:
Do you think we could have a test case to exercise this? Perhaps a simple integration test - like: `ExtHostConfigurationChangedTest.re` Similar in form to: https://github.com/onivim/oni2/blob/master/integration_test/ExtHostCompletionTest.re - except we just need a way to pick up that the extension saw a configuration change.

We could add similar behavior to the `test-extension` - like have it emit some way we can detect the latest value it sees (ie, send a notification) - and then change the configuration, and wait on the notification response.

Alternatively - we do have direct unit tests for the `ExtHostClient` - that's a possibility too: https://github.com/onivim/oni2/blob/9e84aedee6755919a95f11cdf773d4cb5e45b610/test/Extensions/ExtensionClientTest.re#L83

Having a test case that covers this will be helpful when I go to upgrade the extension host, to make sure I don't regress anything

Reply:
Added in https://github.com/onivim/oni2/pull/1489/commits/daf2a3fc734c09e0207f1b7f4241371f58098891